test/eth/TransactionManager.spec.js

```diff
@@ -233,6 +213,38 @@ describe('lifecycle hooks', () => {
     );
   });
 
+  test('finalized Tx is set to correct state without without requiring a call to confirm()', async () => {
+    const openHandlers = makeHandlers('open');
+    txMgr.listen(open, openHandlers);
+    const openTx = txMgr._tracker.get(uniqueId(open));
+
+    await mineBlocks(service);
+
+    expect(openTx.isFinalized()).toBe(true);
+    expect(openHandlers.confirmed).toBeCalled();
+  });
+});
+
+describe('lifecycle hooks errors', () => {
```

Comment:
It is important to note these tests are as a result of web3 error handling when using websockets. When using websockets with these tests, they will fail and terminate automatically as a result of this reject [here](https://github.com/ethereum/web3.js/blob/0d54d94e637832abf97834e4fc444fbff87f874b/packages/web3-utils/src/index.js#L72). The fireError function is called from [here](https://github.com/ethereum/web3.js/blob/0d54d94e637832abf97834e4fc444fbff87f874b/packages/web3-core-method/src/index.js#L364).

I think this is to do with the usage of web3-1.0 where there is an expectation that an error callback or on event handler (like [here](https://web3js.readthedocs.io/en/1.0/web3-eth.html#id64)) is issued to manage a transaction reverting.